projects/core/src/lib/bookmark/bookmark.models.ts

```diff
@@ -1,17 +1,79 @@
-import { MessageType, Message, AnyMessage } from '@bufbuild/protobuf';
+import { Message, AnyMessage, MessageType } from '@bufbuild/protobuf';
 
-export interface BookmarkFragmentProtoDescriptor<T extends Message<T> = AnyMessage> {
-    type: 'proto';
-    proto: MessageType<T>;
-    identifier: string;
+export type BookmarkID = string;
+export type BookmarkType = 'string' | 'binary';
 
-    customSerializer?: {
-       serialize: (x: T) => string | undefined;
-       deserialize: (x: string) => T | undefined;
-    };
+export interface BookmarkFragmentDescriptor {
+  get identifier(): BookmarkID;
+  get type(): BookmarkType;
+  get initialValue(): string | any;
+
+  // Return equality of decoded values
+  equals(one: any, other: any): boolean;
+  serialize(value: any): string | Uint8Array;
+  deserialize(value: string | Uint8Array): any;
```

Comment:
(Are these `any` because of the type erasure happening later on?)

Reply:
Wasn't completely satisfied with this API, it is because `string` fragments are different from `binary` fragments. Will see if the API can be improved.